test/libsolidity/smtCheckerTests/array_members/push_function_return_value.sol

```diff
@@ -0,0 +1,20 @@
+contract C {
+	bytes x;
```

Comment:
I see that `resetReferences()` only looks at the type of the whole variable. What happens if the variable is a struct containing an array and the function returns a reference to that array? Or if it's an array of arrays and one of the nested arrays is returned? Or a mapping where the value type is an array? Mapping nested in a mapping? Can you add test cases covering these?

Reply:
Nested between array/mappings is handled by `SMTEncoder::sameTypeOrSubtype`. Maybe structs should be included there too.

Reply:
Interestingly the example below already works as intended, not sure why.

```
contract C {
	struct S { uint x; uint[] a; }
	S s;
	uint[] b;

	function f(bool c) public {
		require(b.length == 0);
		require(s.a.length == 0);

		b.push();
		assert(b.length == 1); // should hold

		s.a.push();
		assert(s.a.length == 1); // should hold

		uint[] storage ref = s.a;

		ref.push();

		assert(b.length == 1); // should hold but fails due to aliasing
		assert(s.a.length == 2); // should hold but fails due to aliasing

	}
}
```

Reply:
Isn't it because the reference is just a variable and not something returned by a function??